ext/standard/math.c

```diff
@@ -259,16 +258,16 @@ PHP_FUNCTION(abs)
 		Z_PARAM_NUMBER(value)
 	ZEND_PARSE_PARAMETERS_END();
 
-	if (Z_TYPE_P(value) == IS_DOUBLE) {
-		RETURN_DOUBLE(fabs(Z_DVAL_P(value)));
-	} else if (Z_TYPE_P(value) == IS_LONG) {
-		if (Z_LVAL_P(value) == ZEND_LONG_MIN) {
-			RETURN_DOUBLE(-(double)ZEND_LONG_MIN);
-		} else {
-			RETURN_LONG(Z_LVAL_P(value) < 0 ? -Z_LVAL_P(value) : Z_LVAL_P(value));
-		}
-	} else {
-		ZEND_ASSERT(0 && "Unexpected type");
+	switch (Z_TYPE_P(value)) {
+		case IS_LONG:
+			if (Z_LVAL_P(value) == ZEND_LONG_MIN) {
```

Comment:
Possibly as a follow-up wrapping this unlikely condition in ``UNEXPECTED()`` might make sense.

Reply:
This case is a little odd anyway. For 32 bit versions it made sense, because doubles would be able to accurately represent 2147483648, but for 64 bit versions they are not capable of accurately representing 9223372036854775808. I wonder if this should throw instead.

Reply:
Now wrapped in UNEXPECTED, this actually results in slightly better assembly.